app/stores/base/Store.ts

```diff
@@ -182,7 +183,7 @@ export default abstract class Store<T extends Model> {
     }
 
     item.archivedAt = new Date().toISOString();
-    this.add(item);
+    (this as unknown as Store<ArchivableModel>).add(item);
```

Comment:
This is kind of annoying but I don't know of a better way yet. I tried `item.updateData(...)` to circumvent this but that introduces an issue where the archived doc is not conveyed via `wss` to users(other than actor) if that's not in their stores.